src/diffPatch.ts

```diff
@@ -16,7 +16,34 @@ import {
   type SanityPatchMutation,
 } from './patches.js'
 
-const ignoredKeys = ['_id', '_type', '_createdAt', '_updatedAt', '_rev']
+/**
+ * Document keys that are ignored during diff operations.
+ * These are system-managed fields that should not be included in patches on
+ * top-level documents and should not be diffed with diff-match-patch.
+ */
+const SYSTEM_KEYS = ['_id', '_type', '_createdAt', '_updatedAt', '_rev']
+
+/**
+ * Maximum document size to consider for diff-match-patch (1MB)
+ * Based on testing showing consistent 0-2ms performance up to this size
+ */
+const DMP_MAX_DOCUMENT_SIZE = 1_000_000
```

Comment:
"Document" is confusing here since it means something specific in the context of Sanity. Can we rename this to be about strings? Additionally, the timings don't mean much without the hardware it was run on, so maybe just remove that part and phrase it as "consistently fast"?

```suggestion
/**
 * Maximum size of strings to consider for diff-match-patch (1MB)
 * Based on testing showing consistently good performance up to this size
 */
const DMP_MAX_STRING_SIZE = 1_000_000
```